clients/TypeScript/packages/client/src/Connection.ts

```diff
@@ -18,8 +18,19 @@ export interface Connection extends Required<ConnectionConfig> {
 export interface InteractionContext {
   connection: Connection
   socket: WebSocket
+  afterEach: (cb: () => void) => void
 }
 
+/**
+ * Describe how the interaction context behaves. A `LongRunning` context does not close
+ * the underlying connection after a request, it has to be done manually. A `OneTime` context
+ * however will close the connection afterwards.
+ */
+export type InteractionType = (
+  | 'LongRunning'
+  | 'OneTime'
+)
```

Comment:
Now that I've refactored a few things, I also come to doubt the usefulness / relevance of this. Having special logic to handle one-time requests and automatically close connections after sending requests seems a bit superfluous. Rather, this is something that any client's user can do by manually calling `client.shutdown()` when done.

Reply:
The driver of this logic is to enable the use of client-less interactions when making state queries or sending transactions.

Reply:
I see. In that case, I would actually rather define that as a helper function `once` to wrap another query and take care of the resource acquisition for the connection. I'll draft something out.

Reply:
(likely tomorrow or over the weekend :sweat_smile: ....)